compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -121,31 +121,31 @@ where
 
 #[cold]
 #[inline(never)]
-fn mk_cycle<Qcx, V, R>(
+fn mk_cycle<Qcx, V, R, D: DepKind>(
     qcx: Qcx,
-    cycle_error: CycleError,
+    cycle_error: CycleError<D>,
     handler: HandleCycleError,
     cache: &dyn crate::query::QueryStorage<Value = V, Stored = R>,
 ) -> R
 where
     Qcx: QueryContext,
-    V: std::fmt::Debug + Value<Qcx::DepContext>,
+    V: std::fmt::Debug + Value<Qcx::DepContext, D>,
     R: Clone,
 {
     let error = report_cycle(qcx.dep_context().sess(), &cycle_error);
     let value = handle_cycle_error(*qcx.dep_context(), &cycle_error, error, handler);
     cache.store_nocache(value)
 }
 
-fn handle_cycle_error<Tcx, V>(
+fn handle_cycle_error<Tcx, V, D: DepKind>(
     tcx: Tcx,
-    cycle_error: &CycleError,
+    cycle_error: &CycleError<D>,
```

Comment:
```suggestion
    cycle_error: &CycleError<Tcx::DepKind>,
```

You should be able to use it from Tcx here (and probably in a few other places as well).